website/docs/transforms/mock.md

```diff
@@ -26,6 +26,144 @@ transforms:
 
 The example above will replace the resolver of `User.firstName` with a mock that uses `faker.js` to generate a random name.
 
+## Custom mock functions for types
+
+You can mock types with custom mock functions like below;
+
+```yml
+transforms:
+  - mock:
+      mocks:
+        - apply: DateTime
+          custom: graphql-scalars#DateTimeMock 
+          # This will import `DateTimeMock` from `graphql-scalars` for example. Local paths are also supported
+```
+
+## Custom mock functions for fields
+
+You can mock a specific field of a type;
+
+```yml
+transforms:
+  - mock:
+      mocks: 
+        - apply: User.fullName
+          custom: ./user-mocks#fullName
+```
+
+in `user-mocks.js` file;
+
+```js
+module.exports = {
+  fullName: () => 'My Static Name'
+};
+```
+
+## Mocking the lists
+
+By default, Mesh generates two mocked items if the return type is a list. But this can be configured;
+
+Let's say, you have a schema like below;
+
+```graphql
+  type Query {
+    users: [User]
+  }
+  type User {
+    id: ID
+    fullName: String
+  }
+```
+
+Then a configuration like below;
+
+```yml
+transforms:
+  - mock:
+      mocks: 
+        - apply: User.fullName
+          custom: ./user-mocks#fullName
+        - apply: Query.users
+          length: 3
+```
+
+Now `{ users { id fullName } }` query will return 3 of `User` item;
+
+```json
+{
+  "users": [
+    {
+      "id": "SOME_RANDOM_ID",
+      "fullName": "John Doe"
```

Comment:
Actually, since they are mocked by the custom method and not faker, I believe all 3 would return 
```
{
      "id": "SOME_RANDOM_ID",
      "fullName": "My Static Name"
    }
```

Reply:
Yes you're right. It's supposed to be `faker`